### PR TITLE
Fix relative links in main package READMEs so that they work in npm

### PR DESCRIPTION
**Pull request by talldan**
## Description
Follow up to #38605.

In that PR, I discovered when npm displays the README for the package, it interprets relative links incorrectly resulting in a broken URL. This is probably an npm bug, but I haven't figured out where to report it yet (one of these - https://github.com/orgs/npm/repositories?type=all).

This PR updates some of the relative links I found using find/replace, changing them to absolute links.

## Testing Instructions
To reproduce the bug, try the links in the first paragraph of the core-data package in npm - https://www.npmjs.com/package/@wordpress/core-data.

You can see they're broken but they work ok in github - https://github.com/WordPress/gutenberg/tree/trunk/packages/core-data

If you try an absolute link in npm, like the one in the second 'Installation' section for core-data, that seems to work fine - https://www.npmjs.com/package/@wordpress/core-data.

So it seems like changing to absolute URLs is the best thing to do right now.

## Types of changes
Docs

### CODE REVIEW

File: packages/block-serialization-default-parser/README.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Block Serialization Default Parser
 
-This library contains the default block serialization parser implementations for WordPress documents. It provides native PHP and JavaScript parsers that implement the [specification](/docs/contributors/grammar.md) from [`@wordpress/block-serialization-spec-parser`](https://github.com/WordPress/gutenberg/tree/HEAD/packages/block-serialization-spec-parser/README.md) and which normally operates on the document stored in `post_content`.
+This library contains the default block serialization parser implementations for WordPress documents. It provides native PHP and JavaScript parsers that implement the [specification](https://github.com/WordPress/gutenberg/tree/HEAD/docs/contributors/grammar.md) from [`@wordpress/block-serialization-spec-parser`](https://github.com/WordPress/gutenberg/tree/HEAD/packages/block-serialization-spec-parser/README.md) and which normally operates on the document stored in `post_content`.
```
Comment (talldan) [inline packages/block-serialization-default-parser/README.md]: Hmm, this seems to have been a broken link anyway. I'll try to fix it.

File: packages/block-serialization-spec-parser/README.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # Block Serialization Spec Parser
 
-This library contains the grammar file (`grammar.pegjs`) for WordPress posts which is a block serialization [_specification_](/docs/contributors/grammar.md) which is used to generate the actual _parser_ which is also bundled in this package.
+This library contains the grammar file (`grammar.pegjs`) for WordPress posts which is a block serialization [_specification_](https://github.com/WordPress/gutenberg/tree/HEAD/docs/contributors/grammar.md) which is used to generate the actual _parser_ which is also bundled in this package.
```
Comment (talldan) [inline packages/block-serialization-spec-parser/README.md]: Also previously a broken link

File: packages/blocks/README.md [hunk 3]
```diff
@@ -40,7 +40,7 @@ The `enqueue_block_editor_assets` hook is only run in the Gutenberg editor conte
...
+Note that all JavaScript code samples in this document are enclosed in a function that is evaluated immediately afterwards. We recommend using either ES6 modules [as used in this project](https://github.com/WordPress/gutenberg/tree/HEAD/docs/contributors/develop/coding-guidelines.md#imports) (documentation on setting up a plugin with Webpack + ES6 modules coming soon) or these ["immediately-invoked function expressions"](https://en.wikipedia.org/wiki/Immediately-invoked_function_expression) as used in this document. Both of these methods ensure that your plugin's variables will not pollute the global `window` object, which could cause incompatibilities with WordPress core or with other plugins.
```
Comment (talldan) [inline packages/blocks/README.md]: Also this was another broken link

**Review by gziolo — APPROVED** [review]
Yes, this is a very annoying problem. It has two downsides when using full links:
- you can't test new links in PRs (branches)
- when browsing the Block Editor Handbook links will go to GitHub rather than be internal as they used to

**Comment by talldan** [comment]
Thanks for the quick review!

> when browsing the Block Editor Handbook links will go to GitHub rather than be internal as they used to

Hmm, that's a good point and not ideal.

I guess we might be able to rewrite these links when publishing to the handbook, but I'm not sure where that change would be made.

**Comment by gziolo** [comment]
> I guess we might be able to rewrite these links when publishing to the handbook, but I'm not sure where that change would be made.

Yes, it might be possible if always use the same URL. It is now standardized to https://github.com/WordPress/gutenberg/tree/HEAD/ so we can update the regular expression that is used to replace URLs with internal links. I only forgot which repository is used to sync the docs with the dev portal. I'll try to find it.

**Review by kevin940726 — APPROVED** [review]
Awesome work 👍 !

**Comment by gziolo** [comment]
I think I've found the importer:

https://meta.trac.wordpress.org/browser/sites/trunk/wordpress.org/public_html/wp-content/themes/pub/wporg-developer/inc/import-block-editor.php#L241

It should be as simple as ensuring that the regular expressions cover also the case with URLs. Maybe they do already? 😅

**Pull request merged**